configure.ac

```diff
@@ -436,9 +428,22 @@ fi
 
 AM_CONDITIONAL([FPM], [test "x$enable_fpm" = "xyes"])
 
+dnl if the user has specified any CFLAGS, override our settings
 if test "x${enable_dev_build}" = "xyes"; then
    AC_DEFINE(DEV_BUILD,,Build for development)
+   if test "z$orig_cflags" = "z"; then
```

Comment:
ordering of options is relevant here, the generic ones (`-g -O`) should be before the specific stuff (`-f` & `-W`) we have